drivers/can/can_mcux_flexcan.c

```diff
@@ -1166,10 +1170,37 @@ static int mcux_flexcan_init(const struct device *dev)
 
 	DEVICE_MMIO_NAMED_MAP(dev, flexcan_mmio, K_MEM_CACHE_NONE | K_MEM_DIRECT_MAP);
 
+	base = get_base(dev);
+
+#ifdef CONFIG_CAN_MCUX_FLEXCAN_FD
+	/*
+	 * When CAN FD is enabled, FlexCAN RAM can be partitioned into several blocks.
+	 * Each block can accommodate 7 message buffers if payload size is 64-byte.
+	 * Each block can accommodate 32 message buffers if payload size is 8-byte.
+	 * First calculate how many RAM blocks each FlexCAN instance support, then get
+	 * maximum accommodated message buffers under 64-byte payload size.
+	 */
+	max_mb = FSL_FEATURE_FLEXCAN_HAS_MESSAGE_BUFFER_MAX_NUMBERn(base) / 32 * 7;
+#else
+	max_mb = FSL_FEATURE_FLEXCAN_HAS_MESSAGE_BUFFER_MAX_NUMBERn(base);
+#endif /* CONFIG_CAN_MCUX_FLEXCAN_FD */
+
+	if (max_mb < MCUX_FLEXCAN_MAX_RX) {
+		LOG_ERR("Not enough message buffers for RX");
+		return -ENODEV;
+	}
+
+	data->tx_mb_number = max_mb - MCUX_FLEXCAN_MAX_RX;
+	data->rx_mb_number = MCUX_FLEXCAN_MAX_RX;
+
+	if (data->tx_mb_number > MCUX_FLEXCAN_MAX_TX) {
```

Comment:
As tx_mb_number is scaled for canfd I think we need to adopt the MCUX_FLEXCAN_MAX_TX as well don't we?